src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -99,10 +106,127 @@ private static bool TryDecodeSuppressMessageAttributeData (CustomAttribute attri
 
 		private bool IsLocallySuppressed (int id, ICustomAttributeProvider provider, out SuppressMessageInfo info)
```

Comment:
Not for this PR but eventually we should cleanup the types here. The warning requires the "location" to be `IMemberDefinition`, but then all of the suppression logic works on `ICustomAttributeProvider`. Would be nice to unify this.

For example it's possible to suppress a warning at the assembly level, but currently it's not possible to emit a warning at the assembly level.